File: r/src/R/ai/h2o/sparkling/ml/models/H2OBinaryModel.R
H2OBinaryModel.read <- function(path) {
  sc <- spark_connection_find()[[1]]
  jmodel <- invoke_static(sc, "ai.h2o.sparkling.ml.models.H2OBinaryModel", "read", path)
  H2OBinaryModel(jmodel)
}

#' @export H2OBinaryModel
H2OBinaryModel <- setRefClass("H2OBinaryModel", fields = list(
  jmodel = "ANY", modelId = "character"), methods = list(
  initialize = function(jmodel) {
    .self$jmodel <- jmodel
    .self$modelId <- invoke(.self$jmodel, "modelId")
  },
  write = function(path) {
    invoke(.self$jmodel, "write", path)
  }
))

File: py-scoring/src/ai/h2o/sparkling/ml/models/H2OBinaryModel.py
from ai.h2o.sparkling.Initializer import Initializer
from pyspark.ml.param import *
from pyspark.ml.util import _jvm


class H2OBinaryModel:

    def __init__(self, javaModel):
        self._java_obj = javaModel
        self.modelId = javaModel.modelId()

    @staticmethod
    def read(path):
        # We need to make sure that Sparkling Water classes are available on the Spark
        # driver and executor paths
        Initializer.load_sparkling_jar()
        javaModel = _jvm().ai.h2o.sparkling.ml.models.H2OBinaryModel.read(path)
        return H2OBinaryModel(javaModel)

    def write(self, path):
        self._java_obj.write(path)
